core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -14,23 +14,42 @@ import java.util.regex.Pattern
 import akka.actor.{Actor, ActorLogging, ActorRef, Cancellable, DeadLetterSuppression, NoSerializationVerificationNeeded, Props, Status, Terminated}
 import akka.event.LoggingReceive
 import org.apache.kafka.clients.consumer._
-import org.apache.kafka.common.{Metric, MetricName, TopicPartition}
+import org.apache.kafka.common.{Metric, MetricName, PartitionInfo, TopicPartition}
 import org.apache.kafka.common.errors.WakeupException
-
 import java.util.concurrent.locks.LockSupport
 
 import akka.Done
 
 import scala.util.control.{NoStackTrace, NonFatal}
 import scala.collection.JavaConverters._
 import scala.concurrent.duration._
+import scala.util.Try
 
 object KafkaConsumerActor {
   case class StoppingException() extends RuntimeException("Kafka consumer is stopping")
 
   def props[K, V](settings: ConsumerSettings[K, V]): Props =
     Props(new KafkaConsumerActor(settings)).withDispatcher(settings.dispatcher)
 
+  //metadata fetching
+  //NOTE: These block the actor loop!
+  object Metadata {
+    //requests
+    case object ListTopics extends NoSerializationVerificationNeeded
```

Comment:
It would be good to introduce some super trait or class for requests and responses respectively. With that in place, the Metadata handling could easily be kept in a separate receive method.